### PR TITLE
Improve navigation logic, fix imports, and update tests

### PR DESCRIPTION
**Pull request by Raj-G07**
## Description

       Added/fixed unit tests to cover navigation edge cases
       Fixed incorrect/unused imports in several files
## Type of Change

- [x] Bug fix
- [ ] New feature
- [ ] Documentation update
- [ ] Release
- [ ] Refactor
- [ ] Other (please describe):

## Additional Notes

----

> [!IMPORTANT]
> Improve navigation logic in `FrameNavigationManager`, fix imports, and enhance tests for edge cases.
> 
>   - **Navigation Logic**:
>     - Fix off-by-one error in `canGoBack()` and `addToHistory()` in `navigation.ts`.
>     - Adjust history slicing logic in `addToHistory()` to correctly handle forward history.
>   - **Tests**:
>     - Add tests in `navigation.test.ts` to cover edge cases for navigation, including history initialization, duplicate pages, and multiple frames.
>     - Update `helper.test.ts` with mock data for various directory structures and scenarios.
>     - Enhance `env.test.ts` to handle various edge cases in environment variable parsing.
>   - **Imports**:
>     - Fix incorrect/unused imports in `navigation.test.ts` and `helper.test.ts`.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=onlook-dev%2Fonlook&utm_source=github&utm_medium=referral)<sup> for 9b2cf1ca8823792d492368b7625f75d78a7f08a5. You can [customize](https://app.ellipsis.dev/onlook-dev/settings/summaries) this summary. It will automatically update as commits are pushed.</sup>

## Summary by CodeRabbit

- Bug Fixes
  - Improved editor navigation history: forward/back checks adjusted and initial history positioning refined to make frame navigation more consistent and avoid unexpected truncation.

- Tests
  - Migrated navigation tests to the Bun test harness.
  - Reformatted and expanded test inputs for environment parsing and page helpers to better cover quoted values, whitespace, and indentation; no user-facing behavior changes.

### CODE REVIEW

File: apps/web/client/src/components/store/editor/frames/navigation.ts [hunk 1]
```diff
@@ -53,16 +53,16 @@ export class FrameNavigationManager {
...
 
         if (navigationObject.history[navigationObject.currentIndex] === path) {
             return;
         }
 
         // Remove forward history if we're not at the end
-        if (navigationObject.currentIndex < navigationObject.history.length - 1) {
-            navigationObject.history = navigationObject.history.slice(0, navigationObject.currentIndex + 1);
+        if ((navigationObject.history.length - navigationObject.currentIndex) > 1) {
+            navigationObject.history = navigationObject.history.slice(0, navigationObject.currentIndex + 2);
```
[the remark this answers is not in the document]
Comment (Raj-G07) [reply apps/web/client/src/components/store/editor/frames/navigation.ts]: You can fall in this codition if the diff > 1 and if you falls than it means +2 exists and so slice work in this case

File: apps/web/client/src/components/store/editor/frames/navigation.ts [hunk 2]
```diff
@@ -53,16 +53,16 @@ export class FrameNavigationManager {
...
         }
```
[the remark this answers is not in the document]
Comment (Raj-G07) [reply apps/web/client/src/components/store/editor/frames/navigation.ts]: I consider this logic as -1 base indexing not 0 base indexing , so if you consider it , then what your reaction , because during the creation currentIndex = -1  not  0

File: apps/web/client/src/components/store/editor/frames/navigation.ts [hunk 3]
```diff
@@ -24,7 +24,7 @@ export class FrameNavigationManager {
         if (!navigationObject) {
             return false;
         }
-        return navigationObject.currentIndex > 0;
+        return navigationObject.currentIndex > -1;
     }
```
[the remark this answers is not in the document]
Comment (Raj-G07) [reply apps/web/client/src/components/store/editor/frames/navigation.ts]: I consider this logic as -1 base indexing not 0 base indexing , so if you consider it , then what your reaction , because during the creation currentIndex = -1 not 0

File: apps/web/client/src/components/store/editor/frames/navigation.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (Raj-G07) [reply apps/web/client/src/components/store/editor/frames/navigation.ts]: After you look my PR , it is ready to merge , or something else need?

**Comment by Kitenite** [comment]
I hate to be that guy but these are time consuming for me to review so can you not open PRs like this in anymore?
